src/tests/Calculator.test.js

```diff
@@ -0,0 +1,57 @@
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import { render, screen, fireEvent } from '@testing-library/react';
+import Calculator from '../components/Calculator';
+import App from '../App';
+
+const labels = ['AC', 'DEL', '%', '/', '*', '-', '+', '=', '.', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+const ops = ['AC', 'DEL', '%', '/', '*', '-', '+', '=', '.'];
+
+describe('Calculator', () => {
+  test('renders Display component', () => {
+    render(<Calculator />);
+    expect(screen.getByRole('textbox')).toBeInTheDocument();
+  });
+
+  labels.forEach((label) => {
+    test(`renders ${label} Button component`, () => {
+      render(<Calculator />);
+      expect(screen.getByText(label)).toBeInTheDocument();
+    });
+  });
+});
+
+describe('Operations', () => {
+  numbers.forEach((number) => {
+    test(`clicks ${number} Button component`, () => {
+      render(<Calculator />);
+      fireEvent.click(screen.getByText(number));
+      expect(screen.getByRole('textbox').value).toBe(number);
+    });
+  });
+
+  ops.forEach((op) => {
+    test(`clicks ${op} Button component`, () => {
+      render(<Calculator />);
+      const initialValue = screen.getByRole('textbox').value;
+      fireEvent.click(screen.getByText(op));
+      const { value } = screen.getByRole('textbox');
+      if (initialValue !== '0') {
+        expect(value).toBe(op);
+      }
+    });
+  });
+});
```

Comment:
- [ ] Following the project requirements, all the functionalities & scenarios of the operate.js and calculate.js modules should be covered using Jest. Please include tests for calculations made with the calculator (ex: 5 + 6 = 11).

Reply:
Do I also need to test components like home and quotes?

Reply:
Yes, all the components should be tested.